packages/debug/src/browser/console/debug-console-items.tsx

```diff
@@ -191,6 +191,9 @@ export class DebugVariable extends ExpressionContainer {
             this.variablesReference = response.body.variablesReference || 0;
             this.namedVariables = response.body.namedVariables;
             this.indexedVariables = response.body.indexedVariables;
+            if (variablesReference !== 0) {
```

Comment:
not sure it is needed, for example previous value was an object, but new is a string, so then `variableReferenece` will be reset to 0 and we should remove all child variables as well

Reply:
it makes sense